backend_jobs/ingestion_pipeline/providers/image_provider_flickr.py

```diff
@@ -0,0 +1,117 @@
+"""
+  Copyright 2020 Google LLC
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+"""
+
+
+from datetime import datetime
+from pipeline_lib.database_functions import get_doc_by_id
+from pipeline_lib.image_provider_interface import ImageProvider
+from pipeline_lib.additional_classes import ProviderType
+from pipeline_lib.additional_classes import VisibilityType
+from pipeline_lib.additional_classes import ImageAttributes
+import flickrapi
+
+
+class FlickrProvider(ImageProvider):
+    """
+    This class is an implementation for the ImageProvider interface.
+    """
+    def get_images(self, num_of_batches, num_of_images, query_arguments):
+        flickr = flickrapi.FlickrAPI(
+            '2d00397e012c30ccc33ca4fdc05a5c98',
+            'e36a277c77f09fdd',
+            cache=True)
+        photos = flickr.photos.search(text=query_arguments['tag'],
+                     tag_mode = 'all',
+                     tags = query_arguments['tag'],
+                     extras = 'url_c, geo, date_upload, date_taken, original_format,\
+                      owner_name, original_format',
+                     per_page = num_of_images,
+                     page = num_of_batches,
+                     sort = 'relevance')
+        return photos[0]
+
+    def get_num_of_batches(self,query_arguments):
+        photos = self.get_images(1, self.num_of_images,query_arguments)
+        print(photos.attrib)
+        return photos.attrib['pages']
+
+    def get_image_attributes(self, element):
```

Comment:
doc

Reply:
I have a doc in the interface, lint does not require another doc when we inherit.